pkg/controller/servicemesh/member/controller.go

```diff
@@ -341,12 +364,58 @@ func (r *MemberReconciler) updateStatus(member *maistra.ServiceMeshMember, recon
 		Message: message,
 	})
 
+	// TODO: use Client().Status().Patch() and remove the retry code below after we upgrade to controller-runtime 0.2+
 	err := r.Client.Status().Update(context.TODO(), member)
-	if err != nil {
-		return errors2.Wrapf(err, "Could not update status of ServiceMeshMember %s/%s", member.Namespace, member.Name)
-		r.Log.Error(err, "Error updating ServiceMeshMember status")
+	if err == nil {
+		return nil
 	}
-	return nil
+
+	// We only retry on conflict, because a retry will almost certainly succeed, since we first obtain a
+	// fresh instance of the object. We don't retry on any other errors, as it's likely that we'll get the
+	// same error again (plus, it's good to let the human operator know there was an error even if the
+	// retry would succeed).
+	converter := runtime.NewTestUnstructuredConverter(equality.Semantic)
+	for retry := 0; retry < maxStatusUpdateRetriesOnConflict; retry++ {
+		if errors.IsNotFound(err) {
+			// The Member has disappeared, which means it was deleted by someone. We shouldn't treat this as an error
+			// and we shouldn't retry.
+			reqLogger.Info("Couldn't update status, because ServiceMeshMember has been deleted")
+			return nil
+		} else if errors.IsConflict(err) {
+			reqLogger.Info("Ran into conflict when updating ServiceMeshMember's status. Retrying...")
+
+			// This controller owns the ServiceMeshMember's status and can thus always override it (no-one else should
+			// modify the status). We can't simply do a client.Get(), as that would again return the locally cached
+			// object, which means that the update may fail again. We thus need to fetch the object from the API
+			// server directly, update its status, and submit it back to the API Server. Hence the use of Unstructured.
+			freshMember := unstructured.Unstructured{}
+			freshMember.SetAPIVersion(maistra.SchemeGroupVersion.String())
+			freshMember.SetKind("ServiceMeshMember")
+
+			err = r.Client.Get(context.TODO(), types.NamespacedName{member.Namespace, member.Name}, &freshMember)
+			if err != nil {
+				break
+			}
+
+			unstructuredStatus, err := converter.ToUnstructured(member.Status)
```

Comment:
Is `ToUnstructured()` part of this commit, or does it already exist?

Reply:
It's part of apimachinery.